src/components/flat-table/flat-table-row/flat-table-row.component.tsx

```diff
@@ -102,47 +96,94 @@ export const FlatTableRow = React.forwardRef<
     let rowRef = useRef<HTMLTableRowElement>(null);
     if (ref) {
       rowRef = ref as React.MutableRefObject<HTMLTableRowElement | null>;
     }
     const firstColumnExpandable = expandableArea === "firstColumn";
-    const [leftStickyCellWidths, setLeftStickyCellWidths] = useState<number[]>(
-      []
+    const [leftPositions, setLeftPositions] = useState<Record<string, number>>(
+      {}
     );
-    const [rightStickyCellWidths, setRightStickyCellWidths] = useState<
-      number[]
-    >([]);
-    const [leftPositions, setLeftPositions] = useState<number[]>([]);
-    const [rightPositions, setRightPositions] = useState<number[]>([]);
-    const childrenArray = useMemo(() => React.Children.toArray(children), [
+    const [rightPositions, setRightPositions] = useState<
+      Record<string, number>
+    >({});
+    const [firstCellIndex, setFirstCellIndex] = useState(0);
+    const [lhsRowHeaderIndex, setLhsRowHeaderIndex] = useState(-1);
+    const [rhsRowHeaderIndex, setRhsRowHeaderIndex] = useState(-1);
+    const [firstCellId, setFirstCellId] = useState<string | null>(null);
+    const [cellsArray, setCellsArray] = useState<Element[]>([]);
+    const [tabIndex, setTabIndex] = useState(-1);
+    let interactiveRowProps = {};
+
+    useLayoutEffect(() => {
+      const buildPositionMap = (cells: Element[]) =>
+        cells.reduce((acc: Record<string, number>, cell, index) => {
+          const currentId = cells[index].getAttribute("id");
+          if (currentId) {
+            if (index === 0) {
+              acc[currentId] = 0;
+            } else {
+              const previousId = cells[index - 1].getAttribute("id");
+              if (previousId) {
+                acc[currentId] =
+                  acc[previousId] +
+                  (cells[index - 1] as HTMLTableCellElement).offsetWidth;
+              }
+            }
+          }
+          return acc;
+        }, {});
+
+      const cells = rowRef.current?.querySelectorAll("th, td");
+
+      if (cells) {
+        const cellArray = Array.from(cells);
+        setCellsArray(cellArray);
+
+        const firstIndex = cellArray.findIndex(
+          (cell) =>
+            cell.getAttribute("data-component") !== "flat-table-checkbox"
+        );
+        const lhsIndex = cellArray.findIndex(
+          (cell) => cell.getAttribute("data-sticky-align") === "left"
+        );
+        const rhsIndex = cellArray.findIndex(
+          (cell) => cell.getAttribute("data-sticky-align") === "right"
+        );
+
+        if (firstIndex !== -1) {
+          setFirstCellIndex(firstIndex);
+          setFirstCellId(cellArray[firstIndex].getAttribute("id"));
+        }
+        if (lhsIndex !== -1) {
+          setLhsRowHeaderIndex(lhsIndex);
+          const newObj = buildPositionMap(
+            cellArray.slice(0, lhsRowHeaderIndex + 1)
+          );
+
+          if (JSON.stringify(leftPositions) !== JSON.stringify(newObj)) {
```

Comment:
**suggestion** (non-blocking): probably doesn't matter, but I really don't like using `JSON.stringify` to compare 2 objects. I understand why we can't use `==(=)`, but this is effectively only a shallow comparison (as the objects are just a simple map of string IDs to numeric pixel offsets), so you could easily just check the keys are the same and then check equality of values at each key if they are?

Not sure it'll make any realistic difference to performance or anything like that, it just feels better to me in some way I can't even really explain 🤷

Reply:
Good point I'll update